src/linter/report.go

```diff
@@ -471,6 +471,20 @@ function performance_test() {}`,
 			Default: true,
 			Comment: `Report linter error.`,
 		},
+
+		{
+			Name:    "magicMethod",
```

Comment:
I would suggest calling it `magicMethodDecl` as it doesn't check all kinds of mistakes, but only declaration-related ones (visibility, instance/static modifiers).